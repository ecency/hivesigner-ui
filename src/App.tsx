import React, { Suspense } from 'react';
import AppRouter from './router'
import './App.css';

function App() {
  return (
    <React.Fragment>
      <Suspense fallback={null}>
        <AppRouter />
      </Suspense>
    </React.Fragment>
  );
}

export default App;
