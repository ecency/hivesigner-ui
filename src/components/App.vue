<template>
  <div class="text-center">
    <a @click="$emit('select')">
      <span class="mb-2 d-block">
        <Avatar :username="username" :size="60"/>
      </span>
      {{ username }}
    </a>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator'

@Component
export default class App extends Vue {
  @Prop({
    type: String,
    default: '',
  })
  private username!: string
}
</script>
