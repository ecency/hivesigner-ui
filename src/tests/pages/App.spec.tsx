import React from "react";
import App from "../../components/Apps";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
describe("App component Test", () => {
  Enzyme.configure({ adapter: new Adapter() });
  test("Header snapshot testing", () => {
    const header = renderer.create(<div className="slider_header" />).toJSON();
    expect(header).toMatchSnapshot();
  });
  test("slider_logo snapshot testing", () => {
    const logo = renderer.create(<div className="slider_Logo" />).toJSON();
    expect(logo).toMatchSnapshot();
  });
  test("slider_container snapshot testing", () => {
    const slider_container = renderer
      .create(<div className="slider_container" />)
      .toJSON();
    expect(slider_container).toMatchSnapshot();
  });
  test("shapeheader snapshot testing", () => {
    const ShapeHeader = renderer
      .create(<div className="ShapeHeader" />)
      .toJSON();
    expect(ShapeHeader).toMatchSnapshot();
  });
  test("searchContent snapshot testing", () => {
    const SearchContent = renderer
      .create(<div className="SearchContent" />)
      .toJSON();
    expect(SearchContent).toMatchSnapshot();
  });
  test("ProductContainer snapshot testing", () => {
    const ProductContainer = renderer
      .create(<div className="ProductContainer" />)
      .toJSON();
    expect(ProductContainer).toMatchSnapshot();
  });
  test("each components test", () => {
    expect(shallow(<App />).find(".ShapeContainer").length).toEqual(1.5);
    expect(shallow(<App />).find(".ShapeHeader").length).toEqual(1);
    expect(shallow(<App />).find(".StoreTitle").length).toEqual(1);
    expect(shallow(<App />).find(".testing").length).toEqual(1);
  });
});
