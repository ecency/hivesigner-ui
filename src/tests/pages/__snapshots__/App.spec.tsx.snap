// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Test Header snapshot testing 1`] = `
<div
  className="slider_header"
/>
`;

exports[`App component Test ProductContainer snapshot testing 1`] = `
<div
  className="ProductContainer"
/>
`;

exports[`App component Test searchContent snapshot testing 1`] = `
<div
  className="SearchContent"
/>
`;

exports[`App component Test shapeheader snapshot testing 1`] = `
<div
  className="ShapeHeader"
/>
`;

exports[`App component Test slider_container snapshot testing 1`] = `
<div
  className="slider_container"
/>
`;

exports[`App component Test slider_logo snapshot testing 1`] = `
<div
  className="slider_Logo"
/>
`;
